control-plane-operator/controllers/hostedcontrolplane/kas/deployment.go

```diff
@@ -568,19 +569,22 @@ func kasVolumeCloudConfig() *corev1.Volume {
 	}
 }
 
-func buildKASVolumeCloudConfig(configMapName string) func(v *corev1.Volume) {
+func buildKASVolumeCloudConfig(configMapName, providerName string) func(v *corev1.Volume) {
 	return func(v *corev1.Volume) {
-		if v.ConfigMap == nil {
-			v.ConfigMap = &corev1.ConfigMapVolumeSource{}
+		if providerName == azure.Provider {
```

Comment:
Its about unfortunate to need this, but the corev1.LocalObjectReference we pass around doesn't allow us to include typeinfo

Reply:
can't it be always a secret?

Reply:
In the AWS case it doesn't contain anything secret, that is why it is a configmap there

Reply:
yeh so there's aws no need to but it can be a secret so we wouldn't really need to check and basically make everything a secret.